### ISSUE TITLE
Galicaster does not recover correctly from an rtsp error while recording and subsequent recordings fail

### ISSUE THREAD

**Issue by smarquard** [teltek/Galicaster#414]
Galicaster 2.0.x latest code, autorecover = True

Schedule a 5min recording called "recording one"
Schedule another 5min recording after the first called "recording two"

About 30s after "recording one" has started, interrupt the rtsp stream (e.g. reboot the camera).

Galicaster transitions to error state (which is expected), with:

gst-resource-error-quark: Could not read from resource. (9) (gstrtspsrc.c(4806): gst_rtspsrc_loop_udp (): /GstPipeline:galicaster_recorder/galicaster+recorder+bins+rtp+GCrtp:galicaster+recorder+bins+rtp+gcrtp2/GstBin:bin8/GstRTSPSrc:gc-rtp-src:
Could not receive message. (System error))

30 seconds later, autorecover code runs:

recorder/service        Handle recover from error

However, the UI is now in an inconsistent state: top label says "Waiting" while the countdown continues. At the end of the countdown, the UI stays on "Stopping in 00:00".

When the time arrives for "recording two" to start, the state changes to "Recording" BUT the UI still shows "recording one".

At the end of the scheduled time for "recording two", it also does not stop correctly. It seems as Galicaster continues writing to the first recording, and the first recording is ingested (e.g. a 9min30sec recording instead of 2 x 5min recordings).

Expected behaviour:
1. Ideal beheaviour is that a failure during a recording from an rtsp source is not fatal to the recording, e.g. Galicaster catches the error from that pipeline, and stops the camera pipeline but the audio and presentation continue to be recorded and the recording ends normally.
2. Essential behaviour is that a failure during one recording must not prevent subsequent recordings from recording successfully, where the error is recoverable (i.e. by the start time of the second recording, the camera is back online). 

At present, a failure during an earlier recording can cause a whole day of scheduled recordings to fail (until Galicaster is restarted).

**Comment by dpeite** [comment]
Fixed, now when a rtsp stream fails Galicaster stops the recording and recovers correctly.

**Comment by smarquard** [comment]
Thanks! I have tested this on one CA so far. After a failure during a recording, Galicaster successfully returns to preview mode, the failed recording is added to the repository with "Recovery - ..." and the next scheduled recording succeeds.

So the fix is working correctly.

The only "leftover" thing I noticed is that you still get this log line at the original scheduled stop time of the recording:

galicaster      2016-09-06 11:27:59,885 INFO    scheduler/scheduler     Stop record 56946660

although it does not cause any other problems or side-effects.
